### PR TITLE
EGO Gift Extractor

### PR DESCRIPTION
**Pull request by ShinyPulsar**
## About The Pull Request
Adds a one-time use item that extracts EGO gifts. They can be acquired from L-Corp gacha (very rare loot) or from the L-Corp representative with a repeatable 15 PE research.

## Why It's Good For The Game
Because Kirie tasked me to. Ask her, not me.

## Changelog
:cl:
add: Added a new tool : EGO gift extractor.
:cl:

### CODE REVIEW

File: ModularTegustation/tegu_items/representative/research/lcorp.dm [hunk 1]
```diff
@@ -146,6 +146,19 @@ GLOBAL_LIST_EMPTY(lcorp_upgrades)
...
 
+//EGO Gift Extractor
+/datum/data/lc13research/ego_gift_extractor
+	research_name = "Repeatable: EGO Gift Extractor"
+	research_desc = "An order of an EGO Gift Extractor. "
+	cost = AVERAGE_RESEARCH_PRICE+5
+	corp = L_CORP_REP
+
+/datum/data/lc13research/ego_gift_extractor/ResearchEffect(obj/structure/representative_console/caller)
+	if(repeat_cooldown > world.time)
```
Comment (Kitsunemitsu) [inline ModularTegustation/tegu_items/representative/research/lcorp.dm:157]: Give a small notification here and I'll approve it.

This is going to be experimental so....

**Comment by Coxswain-Navigator** [comment]
Non-agent players should never have access to E.G.O gifts under any circumstances.

**Comment by Kitsunemitsu** [comment]
BTW, "Do not merge" is for headcoder use generally. 
I'm going to request this to be changed to Non-clerks, _Or_ security department only.
Pick either or.

**Comment by ShinyPulsar** [comment]
Restricted the tool to Security positions (Agents and ERAs).
Added a notification message from calling in the tool with the Representative's console.

File: ModularTegustation/tegu_items/representative/research/lcorp.dm (hunk 1 above)
Comment (ShinyPulsar) [reply ModularTegustation/tegu_items/representative/research/lcorp.dm:157]: Done.

**Review by Kitsunemitsu — APPROVED** [review]
Approved; Locked to sec roles. Great work.
It's a copypaste job however, giving players something that is technically a debug tool is a horrible idea.

**Pull request merged**